Unittests/Properties/PropertyNullAnnotatorTest.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) rubicon IT GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NullableReferenceTypesRewriter.Properties;
+using NUnit.Framework;
+
+namespace NullableReferenceTypesRewriter.UnitTests.Properties
+{
+  [TestFixture]
+  public class PropertyNullAnnotatorTest
+  {
+    [Test]
+    [TestCase ("[CanBeNull] public string A {get; set;}")]
+    [TestCase ("[CanBeNull] public string A {get;}")]
+    [TestCase (@"[CanBeNull] public string A {get; set;} = """";")]
+    [TestCase (@"[CanBeNull] public string A => """";")]
+    public void PropertyAnnotator_WithCanBeNullAttribute_AnnotatesNullable (string classContent)
+    {
+      var (semantic, syntax) = CompiledSourceFileProvider.CompileInClass ("TestClass", classContent);
+      var annotator = new PropertyNullAnnotator (semantic);
+
+      var annotated = annotator.Visit (syntax);
+
+      Assert.That (annotated.DescendantNodesAndSelf().OfType<PropertyDeclarationSyntax>(), Has.One.Items);
+      var property = annotated.DescendantNodesAndSelf().OfType<PropertyDeclarationSyntax>().Single();
+      Assert.That (property.Type, Is.InstanceOf<NullableTypeSyntax>());
+      Assert.That (((NullableTypeSyntax) property.Type).ElementType.ToString(), Is.EqualTo ("string"));
+    }
+
+    [Test]
+    public void PropertyAnnotator_WithBlockGetterRetursNull_AnnotatesNullable ()
+    {
+      var (semantic, syntax) = CompiledSourceFileProvider.CompileInClass (
+          "TestClass",
+          "private string? _s; public string A {get { if (_s == null) { return null; } return _s; }; set;}");

```

Comment:
What about no explicit `return null;` in the body?

```csharp
private string? _s;
public string A
{
  get { return _s; }
  set { _s = value; }
}
```

Reply:
Does also work, I will add a testcase for it